src/PDPVerifier.sol

```diff
@@ -712,16 +712,24 @@ contract PDPVerifier is Initializable, UUPSUpgradeable, OwnableUpgradeable {
         return BitOps.ctz(index + 1);
     }
 
+    event PriceOracleFailure(bytes reason);
+
     // Add function to get FIL/USD price
-    function getFILUSDPrice() public view returns (uint64, int32) {
+    function getFILUSDPrice() public returns (uint64, int32) {
         // Get FIL/USD price no older than 1 day
-        PythStructs.Price memory priceData = PYTH.getPriceNoOlderThan(
+        try PYTH.getPriceNoOlderThan(
             FIL_USD_PRICE_FEED_ID,
             SECONDS_IN_DAY
-        );
-        require(priceData.price > 0, "failed to validate: price must be greater than 0");
-
-        // Return the price and exponent representing USD per FIL
-        return (uint64(priceData.price), priceData.expo);
+        ) returns (PythStructs.Price memory priceData) {
+            require(priceData.price > 0, "failed to validate: price must be greater than 0");
+            // Return the price and exponent representing USD per FIL
+            return (uint64(priceData.price), priceData.expo);
+        }   catch (bytes memory reason) {
+            // Log issue and fallback on latest unsafe price data
+            emit PriceOracleFailure(reason); 
```

Comment:
We can listen for this event offchain and in the case of StalePrice automatically refresh the Pyth oracle.  It is a simple api call to get the signed data attesting the price you care about and updating the oracle contract.

Reply:
@ZenGround0 So that would be a long running service we host ?

Reply:
Yes anyone who cares about accurate pdp fee calculations.